### PR TITLE
Podcast player: remove wp-components frontend dependency

### PR DESCRIPTION
**Pull request by sgomes**
The podcast player block frontend currently enqueues `wp-components`, which is an absolutely massive 200KB (compressed) dependency, and which comes with other large dependencies of its own.

It does this simply so that it can reuse some SVG-related components. While this is ripe for a deeper refactor, an immediate improvement we can apply is to use `wp-primitives` instead; it has the exact same components (`wp-components` reexports them), and `wp-primitives` has a massively smaller footprint.

Note: I added @Automattic/apex as a reviewer, as I believe that is the team responsible for this block. Please feel free to reassign if that's not the case!

This PR should save ~250KB compressed (!!) on pages with podcast player blocks.

## Proposed changes:
* Podcast player: remove dependency on `wp-components` by replacing it with a dependency on `wp-primitives`

### Other information:

- N/A Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
None, this is a simple performance improvement with no functionality changes.

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
- Create a page with a podcast player block
- Load the page and ensure that `wp-components` is not loaded
- Smoke-test the podcast player block and ensure it works correctly

### CODE REVIEW

File: projects/plugins/jetpack/extensions/blocks/podcast-player/icons/link.js [hunk 1]
```diff
@@ -1,10 +1,7 @@
 /* eslint-disable jsdoc/check-tag-names */
 
-import { SVG, Path } from '@wordpress/components';
+import { SVG, Path } from '@wordpress/primitives';
 
-/**
- * @todo: Replace with `@wordpress/icons` when available
```
Comment (sgomes) [inline projects/plugins/jetpack/extensions/blocks/podcast-player/icons/link.js:6]: Note that we don't actually want to use `@wordpress/icons` here, ever. Doing so would mean pulling in all of the icons to show just a few of them, which would be quite wasteful given that we're using this in the frontend.

**Review by yansern — APPROVED** [review]
Thanks for the simple dependency change that shaves off 250k.
Smoke tested on sandbox using the `jetpack-downloader` command and the player is still working.

**Comment by sgomes** [comment]
Thank you for the review, @yansern! 👍

**Comment by sgomes** [comment]
I believe this needs a @Automattic/jetpack-crew review now, so I added the team as a reviewer.

**Comment by sdixon194** [comment]
Thanks for the cleanup! Tested the podcast player block on JN and a wpcom simple site and the podcast player still works, and wp-components isn't loaded 👍

File: projects/plugins/jetpack/extensions/blocks/podcast-player/icons/link.js (hunk 1 above)
Comment (anomiex) [reply projects/plugins/jetpack/extensions/blocks/podcast-player/icons/link.js:6]: It would depend on the tree shaking, I think. `@wordpress/icons` [is not externalized](https://github.com/WordPress/gutenberg/blob/334dc9c854e8bf4a2758ecda652ffa027e898633/packages/dependency-extraction-webpack-plugin/lib/util.js#L2), so if tree-shaking does a good job of pulling in only the needed icon it might be worthwhile.

Comment (sgomes) [reply projects/plugins/jetpack/extensions/blocks/podcast-player/icons/link.js:6]: Oh, my bad! Do you think it's worth keeping the comment, then?

Comment (anomiex) [reply projects/plugins/jetpack/extensions/blocks/podcast-player/icons/link.js:6]: :shrug: I probably would, since it seems like it's still worth investigating whether the tree-shaking does the right thing. Feel free to reword it to be more explicit about what exactly would need to be investigated.

OTOH, when I tracked down the PR that added the non-externalizing config, I see there was a lot of other discussion and you were involved. So I think you'll know better than I would about all of it. 😀

Comment (sgomes) [reply projects/plugins/jetpack/extensions/blocks/podcast-player/icons/link.js:6]: I'll leave it without the comment, I think, as there isn't a huge benefit to reusing the icon in this case, and introducing the dependency could come with caveats. Thanks!

**Comment by sgomes** [comment]
GitHub isn't letting me merge this just yet, unfortunately. @sdixon194:  I think you may have left your review as a comment instead of an approval? Thanks!

**Review by sdixon194 — APPROVED** [review]

**Comment by sdixon194** [comment]
Strange! Approved again.

**Comment by sgomes** [comment]
Thanks, @sdixon194! Looks like I can merge now 👍
